lib/server-response-adapter.ts

```
@@ -0,0 +1,127 @@
+ * Anthropic's MCP API requires a server response object. This function
+ * creates a fake server response object that can be used to pass to the MCP API.
+ */
+export function createServerResponseAdapter(
+  signal: AbortSignal,
+  fn: (re: ServerResponse) => Promise<void> | void
+): Promise<Response> {
+  let writeHeadResolver: (v: WriteheadArgs) => void
+  const writeHeadPromise = new Promise<WriteheadArgs>(
+    async (resolve, reject) => {
+      writeHeadResolver = resolve
+    }
```

Biome (1.9.4)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)